service/create/domains_test.go

```diff
@@ -0,0 +1,81 @@
+package create
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/giantswarm/awstpr"
+	"github.com/giantswarm/clustertpr"
+	"github.com/giantswarm/clustertpr/kubernetes"
+	"github.com/giantswarm/clustertpr/kubernetes/api"
+	"github.com/juju/errgo"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHostedZoneName(t *testing.T) {
+	tests := []struct {
+		name string
+		tpo  awstpr.CustomObject
+		res  string
+		err  error
+	}{
+		{
+			name: "well-formed domain",
+			tpo: awstpr.CustomObject{
+				Spec: awstpr.Spec{
+					Cluster: clustertpr.Cluster{
+						Kubernetes: kubernetes.Kubernetes{
+							API: api.API{
+								Domain: "api.foobar.example.customer.com",
+							},
+						},
+					},
+				},
+			},
+			res: "example.customer.com",
```

Comment:
It doesn't matter. The point is to have multiple different tests with different values. This is what will happen in production. The business logic has to deal with very different inputs. The test should make sure that this works.